### ISSUE TITLE
Using typeboxes "Index" causes ts error: Type instantiation is excessively deep and possibly infinite

### ISSUE THREAD

**Issue by m1212e** [elysiajs/elysia#353]
```
const someComplexTypeFromMyDatabase= t.Object({
  id: t.String(),
})

export const access = new Elysia()
.post("/blah", async () => {
  return "blah";
}, {
  body: t.Object({
    blah: t.Index(someComplexTypeFromMyDatabase, ["id"])
  })
})
```

results in 

`Type instantiation is excessively deep and possibly infinite.ts(2589)`

**Comment by sinclairzx81** [comment]
@m1212e Hi,

This is actually a TypeBox issue that should be fixed on TypeBox 0.32.0 (next version). 

As a workaround, you can use the following.

[TypeScript Link Here](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgUQDYE8DOwCGAaOeAXzgDMoIQ4ByAUw2x2oChmBjCAO03k0toDClMKloAPACrowtAGIUQAWXQARHDBwAjHJlpwAvIQB0AeU0ArWmxgAKBMzhxgAEwBcxgMowowTgHMbAEo8ZiJA1nFIWDgObngcNjZaTEwDOE5aAHcUBlwg5iNIHhsAIgB6TVQcAAsSgh10TjY4IIMAPkQHOChaGABXKE44EsqakoBuUIJ7R00IZ3R3GFMLK1sZxzhR6vcAbWWASU5ncRs+EEFhUUlpOQVlNQ1tXQJdkpcSgF1Az92ABk+cDKZTgAFVdIRqsBUjAIHBqjgAG56CQeOC0RE4VB9dR6GDVPG3IxdMKhQJAA)

```typescript
import { Elysia, t } from 'elysia'

const someComplexTypeFromMyDatabase = t.Object({
  id: t.String(),
})

export const access = new Elysia()
.post("/blah", async () => {
  return "blah";
}, {
  body: t.Object({
    blah: [t.Index(someComplexTypeFromMyDatabase, ["id"])][0] // Use this to have TS evaluate the type.
  })
})
```

**Comment by m1212e** [comment]
Thank you very much! Also for your work, love typebox!
